mobile/src/full/java/org/openhab/habdroid/core/GcmListenerService.java

```diff
@@ -0,0 +1,84 @@
+/*
+ * Copyright (c) 2010-2018, openHAB.org and others.
+ *
+ *   All rights reserved. This program and the accompanying materials
+ *   are made available under the terms of the Eclipse Public License v1.0
+ *   which accompanies this distribution, and is available at
+ *   http://www.eclipse.org/legal/epl-v10.html
+ */
+
+package org.openhab.habdroid.core;
+
+import android.app.NotificationManager;
+import android.app.PendingIntent;
+import android.content.Intent;
+import android.content.SharedPreferences;
+import android.net.Uri;
+import android.os.Bundle;
+import android.preference.PreferenceManager;
+import android.support.v4.app.NotificationCompat;
+import android.support.v4.content.ContextCompat;
+
+import org.openhab.habdroid.R;
+import org.openhab.habdroid.ui.OpenHABMainActivity;
+import org.openhab.habdroid.util.Constants;
+
+public class GcmListenerService extends com.google.android.gms.gcm.GcmListenerService {
+    static final String EXTRA_NOTIFICATION_ID = "notificationId";
+
+    @Override
+    public void onMessageReceived(String from, Bundle data) {
+        String notificationIdString = data.getString(EXTRA_NOTIFICATION_ID);
+        int notificationId = notificationIdString != null
+                ? Integer.parseInt(notificationIdString) : 1;
+
+        if ("notification".equals(data.getString("type"))) {
+            //for now we use google.sent_time as a time reference for our notifications as the gcm
+            //message does not contain the actual event time from the openhab instance at the moment,
+            //in case gcm time is also missing, we fall back to the reception time which may be delayed (old behaviour)
```

Comment:
I would remove `(old behaviour)` here as it is unclear what "old" is.

Reply:
You're aware that I just moved this and it came in via another PR, right? :-P

But yeah, we surely can fix this here :-)

Reply:
Yes, I'm aware of that. Seems like I missed it in an earlier review.